src/granite/transactions.cr

```diff
@@ -68,7 +68,7 @@ module Granite::Transactions
         end
       {% end %}
     rescue err
-      raise DB::Error.new(err.message)
+      raise DB::Error.new(err.message, cause: err)
```

Comment:
These were originally swallowing the underlying callstacks and only propagating the `cannot allocate memory` error message. Adding `cause` let me find where the root of the problem is (not the why, though, hrm).